### DISCUSSION TITLE
Matching specific DB errors: unique, foreign key and null violations

### DISCUSSION THREAD

**Discussion by Samiisd** [SeaQL/sea-orm#1152] [Q&A] [answered]
Hi,

I am trying to match unique violation error, foreign key violation and not nul violation (for different use cases), but I could not figure out from documentation and source code.

I couldn't a variant of [DbErr](https://docs.rs/sea-orm/latest/sea_orm/error/enum.DbErr.html#) with these errors. So I tried looking into `DbErr::Exec` and `DbErr::Query`, but I am confused between the two as they both "sound alike" to me and take source from `RuntimeError` with a  potential underlying `SqlxError`.

Then, I am not sure either how to get that from the SqlxError. I am using Postgresql as backend.

Is there an example to match these specific errors please?

Thanks

**Comment by billy1624** [comment, marked as answer]
Hey @Samiisd, welcome!! Please upgrade to 0.10.1. We just updated the `DbErr` you can now match the error code thrown by the database.

Example: 
https://github.com/SeaQL/sea-orm/blob/5cce1d3632712fba665553e4264b0e0db9434cf1/tests/crud/error.rs#L13-L55

**Reply by Samiisd** [reply]
Hi @billy1624 ! This is simpler than what I came up with, thanks!

**Reply by tyt2y3** [reply]
We can also extract this parsing result into a enum
